packages/cli/src/ui/components/shared/RadioButtonSelect.tsx

```diff
@@ -4,139 +4,122 @@
-      themeNameDisplay?: string;
-      themeTypeDisplay?: string;
-    }
-  >;
-
-  /** The initial index selected */
+  items: Array<RadioSelectItem<T>>;
   initialIndex?: number;
-
-  /** Function called when an item is selected. Receives the `value` of the selected item. */
   onSelect: (value: T) => void;
-
-  /** Function called when an item is highlighted. Receives the `value` of the selected item. */
   onHighlight?: (value: T) => void;
-
-  /** Whether this select input is currently focused and should respond to input. */
   isFocused?: boolean;
+  showScrollArrows?: boolean;
+  maxItemsToShow?: number;
 }
 
-/**
- * A specialized SelectInput component styled to look like radio buttons.
- * It uses '◉' for selected and '○' for unselected items.
- *
- * @template T The type of the value associated with each radio item.
- */
 export function RadioButtonSelect<T>({
   items,
-  initialIndex,
+  initialIndex = 0,
   onSelect,
   onHighlight,
-  isFocused, // This prop indicates if the current RadioButtonSelect group is focused
+  isFocused,
+  showScrollArrows = true,
+  maxItemsToShow = 10,
 }: RadioButtonSelectProps<T>): React.JSX.Element {
-  const handleSelect = (item: RadioSelectItem<T>) => {
-    onSelect(item.value);
-  };
-  const handleHighlight = (item: RadioSelectItem<T>) => {
-    if (onHighlight) {
-      onHighlight(item.value);
-    }
-  };
+  const [activeIndex, setActiveIndex] = useState(initialIndex);
+  const [scrollOffset, setScrollOffset] = useState(0);
 
-  /**
-   * Custom indicator component displaying radio button style (◉/○).
-   * Color changes based on whether the item is selected and if its group is focused.
-   */
-  function DynamicRadioIndicator({
-    isSelected = false,
-  }: InkSelectIndicatorProps): React.JSX.Element {
-    return (
-      <Box minWidth={2} flexShrink={0}>
-        <Text color={isSelected ? Colors.AccentGreen : Colors.Foreground}>
-          {isSelected ? '●' : '○'}
-        </Text>
-      </Box>
+  useEffect(() => {
+    const newScrollOffset = Math.max(
+      0,
+      Math.min(activeIndex - maxItemsToShow + 1, items.length - maxItemsToShow),
     );
-  }
+    if (activeIndex < scrollOffset) {
+      setScrollOffset(activeIndex);
+    } else if (activeIndex >= scrollOffset + maxItemsToShow) {
+      setScrollOffset(newScrollOffset);
+    }
+  }, [activeIndex, items.length, scrollOffset, maxItemsToShow]);
 
-  /**
-   * Custom item component for displaying the label.
-   * Color changes based on whether the item is selected and if its group is focused.
-   * Now also handles displaying theme type with custom color.
-   */
-  function CustomThemeItemComponent(
-    props: InkSelectItemProps,
-  ): React.JSX.Element {
-    const { isSelected = false, label } = props;
-    const itemWithThemeProps = props as typeof props & {
-      themeNameDisplay?: string;
-      themeTypeDisplay?: string;
-      disabled?: boolean;
-    };
+  useInput(
+    (input, key) => {
+      if (key.upArrow) {
```

Comment:
nit: can you add support for 'j' and 'k' to match the existing ink library we were previously using?